web-components/src/components/form/CreditRetireForm.tsx

```diff
@@ -90,41 +94,75 @@ const sxs = {
 
 export interface BottomCreditRetireFieldsProps {
   mapboxToken: string;
+  arrayPrefix?: string;
+  arrayIndex?: number;
 }
 
+type LocationType = {
+  country?: string;
+  stateProvince?: string;
+  postalCode?: string;
+};
+
 export const BottomCreditRetireFields: React.FC<BottomCreditRetireFieldsProps> =
-  ({ mapboxToken }) => {
+  ({ mapboxToken, arrayPrefix, arrayIndex }) => {
     const styles = useStyles();
-    const {
-      values: { country, stateProvince, postalCode },
-      setFieldValue,
-    } = useFormikContext<RetireFormValues>();
+    const { values, setFieldValue } = useFormikContext<
+      RetireFormValues | RetireFormValuesArray
+    >();
+
+    const item =
+      typeof arrayIndex === 'number'
+        ? (values as RetireFormValuesArray).recipients[arrayIndex]
+        : (values as RetireFormValues);
+
+    const { country, stateProvince, postalCode } = item;
 
     useEffect(() => {
-      const setRetirementLocation = async (): Promise<void> => {
-        const isoString = await getISOString(mapboxToken, {
-          countryKey: country,
-          stateProvince,
-          postalCode,
-        });
-        setFieldValue('retirementLocation', isoString);
+      const retirementLocationName = arrayPrefix
+        ? `${arrayPrefix}retirementLocation`
+        : 'retirementLocation';
```

Comment:
maybe we could create a util function for that since this same check is done several times in this file